src/de/hhn/aib/labsw/blackmirror/controller/PageController.java

```diff
@@ -0,0 +1,124 @@
+package de.hhn.aib.labsw.blackmirror.controller;
+
+import de.hhn.aib.labsw.blackmirror.view.widgets.AbstractWidget;
+
+import java.util.ArrayList;
+
+/**
+ *  PageController Class to hold all pages and provide all necessary methods.
+ */
+public class PageController
+{
+    private ArrayList<Page> pages;
+    private int pageIndex = 0;
+    private boolean isStandby = false;
+
+    public PageController ()
+    {
+        pages = new ArrayList<>();
+
+    }
+
+    /**
+     * Creates a new Page and adds it to the list of all Pages.
+     * @param widgetsOnPage Widgets to be shown on the new page.
+     */
+    protected void addPage(ArrayList<AbstractWidget> widgetsOnPage)
+    {
+        pages.add(new Page(widgetsOnPage));
+    }
+
+    /**
+     * Adds a new page at the given index.
+     * @param widgetsOnPage Widgets to be shown on the new page.
+     * @param pageIndex Index where the page should be.
+     */
+    protected void addPageAtIndex(int pageIndex, ArrayList<AbstractWidget> widgetsOnPage)
+    {
+        pages.add(pageIndex, new Page(widgetsOnPage));
+    }
+    /**
+     * Deletes a page at the given index.
+     * @param pageIndex Index of the page to be deleted.
+     */
+    protected void deletePage (int pageIndex)
+    {
+        pages.remove(pageIndex);
+    }
+
+    /**
+     * Moves a page to another position in the list.
+     * @param currentPosition Index, where the page currently is.
+     * @param newPosition Index, where the page should be moved to.
+     */
+    protected void movePage (int currentPosition, int newPosition)
```

Comment:
Perhaps replace the index with the reference (like so 'protected void movePage(Page page, int newPosition') or rename 'currentPosition' to 'pageIndex'.
In my opinion usually prefer the reference over the index of this reference.